lib/puppet/pops/evaluator/deferred_resolver.rb

```diff
@@ -91,6 +97,12 @@ def resolve_futures(catalog)
           unless r.sensitive_parameters.include?(k.to_sym)
             r.sensitive_parameters = (r.sensitive_parameters + [k.to_sym]).freeze
           end
+        # If the value is a DeferredValue and it has an argument of type PSensitiveType, mark it as sensitive
+        # The DeferredValue.resolve method will unwrap it during catalog application
+        elsif resolved.is_a?(Puppet::Pops::Evaluator::DeferredValue)
+          if v.arguments.any? {|arg| arg.is_a?(Puppet::Pops::Types::PSensitiveType)} and not r.sensitive_parameters.include?(k.to_sym)
```

Comment:
One tricky issue is deferred values can nest like:
https://github.com/puppetlabs/puppet/blob/c4c07ee6aa088324d9d8612c9fdb1daed75a739d/spec/unit/pops/evaluator/deferred_resolver_spec.rb#L35-L38
https://github.com/puppetlabs/puppet/blob/c4c07ee6aa088324d9d8612c9fdb1daed75a739d/spec/unit/functions/call_spec.rb#L115-L118

I think this will approach will still work (?) as the `resolve_futures` method is called recursively with each `DeferredValue` and if any of them have `PSensitiveType` arguments, then the overall parameter will be marked as sensitive (but only marked once). But I'd like to write a test to confirm that happens.